### PR TITLE
Guideline updates

### PR DESCRIPTION
**Pull request by muffato**
Added:

- A description of the branch model and how to deal with bugfixes
- Instructions for adding a new pipeline.
   - Can you double-check who can add repositories to the nextflow_all and nextflow_admin teams ? I couldn't find clear documentation on GitHub
   - Also, for updating the settings via the pipeline website, I know it's something people can do on their own, but it needs the pipeline to be recognised by the website first. Should we change the instruction to "Wait a few hours until the pipeline shows up, then click "Fix data" ?

### CODE REVIEW

**Comment by muffato** [comment]
The docs are deployed at http://pipelines.tol-dev.sanger.ac.uk/docs

**Comment by muffato** [comment]
Just added a small paragraph about the Zenodo integration. Even if it's already mentioned in the "Release instructions", better to do it right at the beginning before anyone forgets

File: markdown/developers/create-pipeline.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+All pipelines _must_ use the [nf-core template](https://nf-co.re/docs/contributing/guidelines/requirements/use_the_template).
+This is done by using the `nf-core create` command - see [the docs](https://nf-co.re/tools#creating-a-new-pipeline) for detailed instructions.
+This tool does lots of things for you: it gives you the correct file structure and boiler plate code
+and also sets up the required `git` infrastructure for you to keep your pipeline in sync in the future.
+
+## Push to GitHub
+
+Create an empty repository on GitHub for your new pipeline under <https://github.com/sanger-tol>.
+Do this by going to the GitHub website and clicking + then _New Repository_.
+
```
Comment (gq1) [inline markdown/developers/create-pipeline.md:21]:
```suggestion
Make sure the new repository is under sanger-tol organisation, not under your own name or other orgisation. 
```

File: markdown/developers/create-pipeline.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+This page is heavily inspired by the nf-core page [Adding a new pipeline](https://nf-co.re/docs/contributing/adding_pipelines).
+
+## Create the pipeline
+
+All pipelines _must_ use the [nf-core template](https://nf-co.re/docs/contributing/guidelines/requirements/use_the_template).
+This is done by using the `nf-core create` command - see [the docs](https://nf-co.re/tools#creating-a-new-pipeline) for detailed instructions.
+This tool does lots of things for you: it gives you the correct file structure and boiler plate code
+and also sets up the required `git` infrastructure for you to keep your pipeline in sync in the future.
+
```
Comment (gq1) [inline markdown/developers/create-pipeline.md:16]: Please use `sanger-tol` as the pipeline prefix and you may skip some of template areas.

Comment (gq1) [reply markdown/developers/create-pipeline.md:16]: Currently only `iGenomes config` needs to be skipped for some pipelines, not others.

What about the multiqc? no options to skip but we don't want it for some pipelines and it is tedious to remove it later.

Comment (muffato) [reply markdown/developers/create-pipeline.md:16]: Very good point. I completely forgot about the configuration options in `nf-core create`. I've added that now.

I don't want to put removing multiqc in the instructions. Yes it's very tedious to remove completely and properly, so I think the easier approach is to simply keep the module around and the various functions and template files, but. not use it in any workflow / sub-workflow. This also simplifies template update as git won't flag those files as merge conflicts.

**Comment by muffato** [comment]
I've addressed both comments and updated the checkout on the dev VM

File: markdown/developers/create-pipeline.md [hunk 3]
```diff
@@ -14,6 +14,14 @@ This is done by using the `nf-core create` command - see [the docs](https://nf-c
 This tool does lots of things for you: it gives you the correct file structure and boiler plate code
 and also sets up the required `git` infrastructure for you to keep your pipeline in sync in the future.
 
+When asked _Do you want to customize which parts of the template are used ?_, answer `Y`.
```
Comment (gq1) [inline markdown/developers/create-pipeline.md:17]: Let's use small `y` for yes. Big `N `is the default answer.

**Comment by gq1** [comment]
I don't have permission to add any repos to teams.

I think the current instruction to fix the repo setting is fine.  The statistics page is only updated once in the morning by the cron job, and the same as the pipeline list. The developers don't really need to worry about this at the beginning, we can do this behind.

**Review by gq1 — APPROVED** [review]

**Pull request merged**